src/utils/forms.js

```diff
@@ -0,0 +1,17 @@
+const FORM_URLENCODED = 'application/x-www-form-urlencoded'
+
+function collectRequestBody(request, callback) {
```

Comment:
I think you might want to use a library to read `request` body instead? Libraries like `raw-body` adds lots of tricks and handle some edge cases when doing this.

```js
const getRawBody = require('raw-body')
const body = String(await getRawBody(request))
```

Reply:
Oh nice, thanks!